draft-ietf-capport-architecture.xml

```diff
@@ -695,7 +695,29 @@ o . . . . . . . . . . . . . . . . . . .| |. . . . . . . . . . . o
                 link. In such cases, identification could be performed by subnet, such as the /64
                 to which the IP belongs.
             </t>
-           </section>
+         </section>
+         <section anchor="context_free_url" title="Context-free URL">
+           <t>
+             The URLs provided in the API SHOULD contain all the information necessary to render
+             the resources requested: they should be context free so it does not depend on ambient
+             information, such as remote address on the connection. This is to ensure that User
+             Equipment can access URLs from a network other than the one that contains the captive
```

Comment:
Given my understanding (as outlined above), I think the issue isn't that the UE can't access URLs in a network other than the one in which they were initially captive, but rather that the resource at that URL may not render correctly in the new network. Perhaps some slight rephrasing could capture this?

Reply:
I could be mistaken, but my understanding of "access from another network" was that it could include something like a cell phone that tried to access the URL via the cell network and not via the captive wifi network.

Reply:
Yeah, that makes sense -- that's an entirely plausible scenario. I thought that the problem was that if the device serving the content at the URL used information from its network stack to render the content, then it would not work in this case.

E.g. imagine that it determined the identity of the UE based on the source ip address of the tcp connection over which the http request was carried. That would almost certainly fail if the UE switched from a captive wifi network to a cell-phone network. However, if the information the UE used to identify itself to the device in the first place was carried in the URL, this would not be a problem.

Reply:
My reading of the current text is that it's saying that we want to put to context into the URL so that the URL is still accessible. But, if we follow the advice of using a public DNS name for the host, it should always be accessible. At issue is the content it serves being meaningful.

Reply:
I guess we have different interpretations of context

personally, I would not call anything in the URL as the context of the URL

for example, https://www.google.com/?q=ietf, the search query is part of the URL (and I would not call that context of the url)

I think the current text is stating that the resource behind the URL should be accessible even if a non-captive network is used to access the URL. 

Shall we clarify what context means here? or what do you suggest rephrase please?

Reply:
I think we agree on what the context is. Just for good measure, I'll give an example. :)

For example, if the API returned https://example.com/captive-portal for its `user-portal-url`, and example.com used the source-ip address of the connection to present quota information to the user while rendering that page, we'd have a problem if the user briefly switched to an LTE network from the captive wifi network. 

However, if the API returned https://example.com/captive-portal?ue-identifier=X, and example.com used X to look up and present the quota information, then it wouldn't matter which network the UE was connected to.

My concern was with the phrasing:

> This is to ensure that User Equipment can access URLs from a network other"

The question of whether the URL is *accessible* is independent of whether the content served at that URL is context free. What matters here is whether it's *meaningful* and *correct*. So, maybe rephrase to:

"This is to ensure that the content served from these URLs is correct and meaningful to the User Equipment, even when accessed from a network other than the one that contains the captive portal".

Sorry about the confusion and ramblings. Being somewhat disconnected from the WG at the moment, I just want to make sure I'm not misunderstanding. :)